networkx/drawing/nx_pylab.py

```diff
@@ -51,6 +53,1091 @@
+            "bbox": None,
+        },
+        "node_shape": "o",
+        "node_alpha": 1.0,
+        "node_border_width": 1.0,
+        "node_border_color": "face",
+        "edge_visible": True,
+        "edge_width": 1.0,
+        "edge_color": "#000000",
+        "edge_label": {
+            "size": 12,
+            "color": "#000000",
+            "family": "sans-serif",
+            "weight": "normal",
+            "alpha": 1.0,
+            "bbox": {"boxstyle": "round", "ec": (1.0, 1.0, 1.0), "fc": (1.0, 1.0, 1.0)},
+            "h_align": "center",
+            "v_align": "center",
+            "pos": 0.5,
+            "rotate": True,
+        },
+        "edge_style": "-",
+        "edge_alpha": 1.0,
+        "arrowstyle": "-|>" if G.is_directed() else "-",
+        "arrowsize": 10 if G.is_directed() else 0,
+        "edge_curvature": "arc3",
+        "edge_source_margin": 0,
+        "edge_target_margin": 0,
+        "hide_ticks": True,
+    }
+
+    # Check arguments
+    for kwarg in kwargs:
+        if kwarg not in defaults:
+            raise nx.NetworkXError(
+                f"Unrecongized visualization keyword argument: {kwarg}"
+            )
+
+    if canvas is None:
+        canvas = plt.gca()
+
+    if kwargs.get("hide_ticks", defaults["hide_ticks"]):
+        canvas.tick_params(
+            axis="both",
+            which="both",
+            bottom=False,
+            left=False,
+            labelbottom=False,
+            labelleft=False,
+        )
+
+    ### Helper methods and classes
+
+    # I hope that the order-perserving nature of dicts in python 3.7+
+    # means that G.nodes() will always return the same order.
+    def node_property_sequence(seq, attr, default=None):
+        """Return a sequence of attribute values for the given sequence, using default if not None"""
+
+        # If the attr isn't actually a graph attr, but was explicitly passed as an argument it must be a
+        # user-default value. Allow None to be used to tell draw to skip attributes which are on the graph
+        if (
+            attr is not None
+            and nx.get_node_attributes(node_subgraph, attr) == {}
+            and any(attr == v for k, v in kwargs.items() if "node" in k)
+        ):
+            return [attr for _ in seq]
+
+        # The inner function isn't my first choice here, but if I use a regular try-except
+        # around the asarray call I can't seem to isolate which node is problematic for
+        # the error message.
+        def error(n, attr, default):
+            if default is None:
+                raise nx.NetworkXError(
+                    f"Node {n} is missing required attribute: {attr}"
+                )
+            else:
+                return False
+
+        return [
+            (
+                node_subgraph.nodes[n][attr]
+                if attr in node_subgraph.nodes[n] or error(n, attr, default)
+                else default
+            )
+            for n in seq
+        ]
```

Comment:
I think this is simpler and easier to read, but that may just be me
```suggestion
        if default is None:
            # raise instead of using default value
            for n in seq:
                if attr not in node_subgraph.nodes[n]:
                    raise nx.NetworkXError(f"Attribute {attr} missing for node {n}")

        return [node_subgraph.nodes[n].get(attr, default) for n in seq]
```

Reply:
This is indeed much cleaner and readable. I also don't know why I didn't realize that I could call `node_subgraph.nodes[n].get`...